pkgs/desktops/deepin/dde-file-manager/default.nix

```diff
@@ -0,0 +1,235 @@
+{ stdenv, fetchFromGitHub, fetchpatch, pkgconfig, qmake, qttools,
+  qtmultimedia, qtx11extras, gnome3, gsettings-qt, dtkcore, dtkwidget,
+  deepin-movie-reborn, jemalloc, ffmpegthumbnailer, poppler, taglib,
+  libsecret, kcodecs, deepin-gettext-tools, dde-polkit-agent, deepin,
+  qt5integration, gtk2, qtsvg, file, avfs, polkit, polkit-qt,
+  deepin-shortcut-viewer, xdg-user-dirs, dde-qt-dbus-factory,
+  deepin-terminal, gst-plugins-good, mpv, zlib, libX11, gvfs, glib,
+  xorg, udisks2-qt5, dde-daemon, dde-dock, deepin-anything,
+  autoPatchelfHook }:
+
+stdenv.mkDerivation rec {
+  name = "${pname}-${version}";
+  pname = "dde-file-manager";
+  version = "4.8.4";
+
+  src = fetchFromGitHub {
+    owner = "linuxdeepin";
+    repo = pname;
+    rev = version;
+    sha256 = "0yhy36gkk5nyn56sn1v3hjri22nxs9k7raflbszahm574xpr1akb";
+  };
+
+  nativeBuildInputs = [
+    deepin.setupHook
+    qmake                   # build
+    qttools                 # build
+    pkgconfig               # build
+    deepin-gettext-tools    # build
+    autoPatchelfHook
+  ];
+
+  buildInputs = [
+    avfs                   # recommended
+    dde-dock                # build
+    dde-polkit-agent
+    dde-qt-dbus-factory    # build arch ?
+    deepin-anything        # build run [amd64 i386]
+    deepin-movie-reborn      # build arch
```

Comment:
I think they need this for `dde-video-preview-plugin` and  properly `libdmr` in `deepin-movie-reborn`.

With that setup we should make `deepin-movie-reborn` a split package/multi output derivation I think.

Reply:
> I think they need this for `dde-video-preview-plugin` and properly `libdmr` in `deepin-movie-reborn`.
> 
> With that setup we should make `deepin-movie-reborn` a split package/multi output derivation I think.

I didn't understand well what you mean here.

Reply:
`outputs = [ "out" "dev" ];` for `deepin-movie-reborn`.

Reply:
Talking about `deepin-movie-reborn`:

In [Void Linux](https://github.com/void-linux/void-packages/tree/master/srcpkgs/deepin-movie-reborn) there are the base and the devel packages. The devel has:
- `usr/lib/pkgconfig`
- `usr/include`
- `usr/lib/*.so`

In [Debian](https://salsa.debian.org/pkg-deepin-team/deepin-movie-reborn/tree/master/debian) there are 3 packages:
- **deepin-movie**: Deepin movie player (_depends on libdmr0.1_)
  - `usr/bin/deepin-movie`
  - `usr/share/applications/deepin-movie.desktop`
  - `usr/share/deepin-movie/translations/*.qm`
  - `usr/share/icons/*`
- **libdmr-dev**: Deepin movie player - widget library (development files) (_depends on libdmr0.1_)
  - `usr/include/libdmr/*.h`
  - `usr/lib/*/libdmr.so`
  - `usr/lib/*/pkgconfig/libdmr.pc`
- **libdmr0.1**: Deepin movie player - widget library 
  - `usr/lib/*/libdmr.so.*`

NixOS with `outputs = [ "out" "dev" ]` automatically gives for the `dev` output:
  - `usr/include/libdmr/*.h`
  - `usr/lib/pkgconfig/libdmr.pc`

If I have understood correctly, the idea is to split `deepin-movie-reborn` so that `dde-file-manager` (which uses `libdmr` from `deepin-movie-reborn`) can be compiled and run without requiring the full `deepin-movie-reborn` package.

@worldofpeace, in face of this, is it more reasonable to have `outputs = [ "out" "lib" "dev" ]`, with `out` depending on `lib`? If so, how to expresses the runtime dependency of `out` on `lib`?

Then `dde-file-manager` would have `dev` in `nativeBuildInputs` and `lib` in `buildInputs`.

Or should I do something different from what I mentioned above?

Reply:
> `outputs = [ "out" "dev" ];` for `deepin-movie-reborn`

Done.